include/seqan3/argument_parser/validators.hpp

```diff
@@ -235,57 +237,48 @@ value_list_validator(std::vector<const char *> const & v) -> value_list_validato
 value_list_validator(std::initializer_list<const char *> const & v) -> value_list_validator<std::string>;
 //!\}
 
-/*!\brief A validator that checks if a filenames has one of the valid extensions.
+/*!\brief An abstract base class for the file and directory validators.
  * \ingroup argument_parser
- * \implements seqan3::validator_concept
  *
  * \details
  *
- * On construction, the validator must receive a list (vector) of valid file extensions.
- * The struct than acts as a functor, that throws a seqan3::parser_invalid_argument
- * exception whenever a given filename (string) is not in the given extension list.
- *
- * \snippet test/snippet/argument_parser/validators_3.cpp usage
- *
- * \note The validator works on every type that can be implicitly cast to std::filesystem::path.
+ * This class provides a common interface for seqan3::input_file_validator and the seqan3::output_file_validator as
+ * well as the seqan3::input_directory_validator and seqan3::output_directory_validator.
  */
-class file_ext_validator
+class file_validator_base
 {
 public:
-    //!\brief Type of values that are tested by validator
+
+    //!\brief Type of values that are tested by validator.
     using value_type = std::string;
 
-    /*!\brief Constructing from a vector.
-     * \param[in] v The vector of valid file extensions to test (e.g. {"fa", "fasta"}).
-     * \param[in] c Case sensitivity flag. Set true for case sensitivity. Default: false (case insensitive).
-     *
-     * For case insensitivity, everything is converted to lower case characters.
+    /*!\name Constructors, destructor and assignment
+     * \{
      */
-    file_ext_validator(std::vector<std::string> const & v, bool const c = false) :
-        case_sensitive{c}
-    {
-        extensions = c ? v : std::vector<std::string>{v | view::to_lower};
-    }
-
-    /*!\brief Tests whether the filepath \p path ends with a valid extension.
-     * \param path The input value to check.
-     * \throws parser_invalid_argument
+    file_validator_base() = default;                                        //!< Defaulted.
+    file_validator_base(file_validator_base const &) = default;             //!< Defaulted.
+    file_validator_base(file_validator_base &&) = default;                  //!< Defaulted.
+    file_validator_base & operator=(file_validator_base const &) = default; //!< Defaulted.
+    file_validator_base & operator=(file_validator_base &&) = default;      //!< Defaulted.
+    virtual ~file_validator_base() = default;                               //!< Virtual destructor.
```

Comment:
why?

Reply:
Well, that's the world of runtime polymorphism. 

> Deleting a derived class object using a pointer to a base class that has a non-virtual destructor results in undefined behavior. To correct this situation, the base class should be defined with a virtual destructor.